### ISSUE TITLE
X20Pro AW bootloader 

### ISSUE THREAD

**Issue by Alex00101** [FrSkyRC/ETHOS-Feedback-Community#3791]
I'm sorry if it is already covered or it's a user error but I'm experiencing issues with X20 Pro AW bootloader I never had with X10Express. Bootloader version 1.5.0.
1. 1st Ethos upgrade. Connected to PC in bootloader mode, Suite detected new firmware, system hanged on "writing firmware" (firmware.bin was copied correctly). Disconnected and power cycled radio to the bootloader mode, bootloader displays "writing firmware" and hangs. Somehow managed to update to 1.5.4 but it involved multiple attempts, disconnecting battery, cleaning some garbage files that were somehow created etc.
2. Suite refuses to switch between Ethos, bootloader and serial modes, in particular I cannot enter lua debug session. The error is "your bootloader does not support mode switching ".
3. Not bootloader related but model creation wizard enforced on the 1st run when there is no models (after "read error" message) ends in emergency mode. The model is created however and seems to be working on subsequent runs.

Appreciate any clarity on this.  Thanks

**Comment by bsongis-frsky** [comment]
I should receive an X20-PRO-AW in a couple of days. I will be able to check this. I am not able to replicate any of the points above on a X20-PRO (not AW)

**Comment by bsongis-frsky** [comment]
Are you able to reproduce the Point3 after you remove the models directory?

**Comment by Alex00101** [comment]
Thanks. Yes when models directory is empty, on start I'm getting "read error, starting wizard" message, then when I click the last arrow on the wizard I'm getting EM. After power cycle the model (and the models dir) is there and everything works

**Comment by Alex00101** [comment]
One thing, when I connect the radio to the pc, (I have no advantage card installed) I see 2 drives (flash and radio), not one. Not sure if this is relevant. Firmware.bin got copied to Radio

**Comment by bsongis-frsky** [comment]
It's normal that you see 2 drives (flash + nand)
With an SD you would have 3 drives

**Comment by bsongis-frsky** [comment]
> Thanks. Yes when models directory is empty, on start I'm getting "read error, starting wizard" message, then when I click the last arrow on the wizard I'm getting EM. After power cycle the model (and the models dir) is there and everything works

Even with 1.5.4 ? And the latest nightlies?

**Comment by Alex00101** [comment]
I'm on 1.5.4 and going to retest again on the other radio (X10E). I haven't tried nighties yet.

**Comment by Alex00101** [comment]
Update. Refreshed bootloader from github, reflashed 1.5.4. Now when I delete models folder, model creation wizard does not seem to cause EM anymore. Also was able to switch to lua debug (serial) mode at least when the radio was connected to PC in bootloader mode.

**Comment by bsongis-frsky** [comment]
I have received the radio today, I will start updating it tomorrow

**Comment by Alex00101** [comment]
Thanks a lot!

**Comment by bsongis-frsky** [comment]
> 1st Ethos upgrade. Connected to PC in bootloader mode, Suite detected new firmware, system hanged on "writing firmware" (firmware.bin was copied correctly). Disconnected and power cycled radio to the bootloader mode, bootloader displays "writing firmware" and hangs. Somehow managed to update to 1.5.4 but it involved multiple attempts, disconnecting battery, cleaning some garbage files that were somehow created etc.

I haven't had any issue on upgrade. I think you had an issue when copying the file and it was corrupted on your SD. Ethos Suite should perhaps ensure that the file is perfectly copied before launching the upgrade ...

**Comment by Alex00101** [comment]
Thank you, makes sense. Yes checking the file for integrity before starting
upgrade is a great idea

**Comment by Alex00101** [comment]
P.s. I didn't and still don't have an SD in AW.

**Comment by bsongis-frsky** [comment]
> 2. Suite refuses to switch between Ethos, bootloader and serial modes, in particular I cannot enter lua debug session. The error is "your bootloader does not support mode switching ".

Reproduced, same issue here. The drives are correctly mounted. Issue in Ethos Suite. It waits forever.

It is 100% reproductible. For me it's impossible to use the DEV move in Ethos Suite

**Comment by bsongis-frsky** [comment]
> Not bootloader related but model creation wizard enforced on the 1st run when there is no models (after "read error" message) ends in emergency mode. The model is created however and seems to be working on subsequent runs.

Perhaps my radio was different? I haven't had to calibrate it, and there was one model on it already. Unless my children played with it yesterday, which is very possible, as they know the radio very well and this one looks better than all the others with the wires going outside ;)

**Comment by Alex00101** [comment]
Hmmm looks like you've got a "used" one 😀

**Comment by Alex00101** [comment]
That issue may be related to #1 which is corrupted flash filesystem after
failed firmware upgrade

**Comment by bsongis-frsky** [comment]
> Not bootloader related but model creation wizard enforced on the 1st run when there is no models (after "read error" message) ends in emergency mode. The model is created however and seems to be working on subsequent runs.

I have tried to remove radio.bin and the models directory to start again as if the radio was unused, and nothing, everything works perfectly

**Comment by Alex00101** [comment]
That is consistent with my experience, when I scanned the flash drive,
removed the garbage,  and reflashed the firmware. After that even after
removing models directory etc everything worked. So I might had a once off
issue.
My apologies for a likely false alarm.
At least the bootloader issues seems reproducible so I'm not entirely
wasting your time (hope)

**Comment by bsongis-frsky** [comment]
Why do you say it's a bootloader issue? I would say it's Ethos Suite!

**Comment by bsongis-frsky** [comment]
This radio is damn beautiful, and it looks as heavy as I like, it's not a toy!

**Comment by Alex00101** [comment]
Sorry I thought it was. It very well may be the Ethos suite, I jumped into
conclusion.

Yes I like this radio. Best I ever had, although I'm yet to fly with it.

**Comment by bsongis-frsky** [comment]
Relevant part of this issue are merged into #3822, I will close it
